src/services/sync/strategies/TSP/TSPSchoolSyncer.js

```diff
@@ -125,8 +128,25 @@ class TSPSchoolSyncer extends mix(Syncer).with(ClassImporter) {
 			);
 			this.logInfo('Done.');
 		}
+		await Promise.all(
+			schools.map((school) => schoolLimit(() => this.processSchool(school, teacherMap, studentMap, classMap)))
```

Comment:
1. In my opinion, reinventing the wheel, i.e., writing code from scratch to limit the number of queries will only add unnecessary work to us
2. All uses are surrounded by a try catch block, so it should not throw an error anywhere
3. Maybe first, instead of testing an external library, we should start with testing our own code, because I do not see any tests for TSPSchoolSyncer (maybe a new ticket can be created for this?)